bms_factors/charge_rate.py

```diff
@@ -0,0 +1,23 @@
+class Charge_Rate:
+    def __init__(self) -> None:
+        self.max_val = 0.8
+        self.min_val = 0.0
+        self.print_msg = lambda str, verbosity : print(str) if verbosity else None
+    
+    def verify_charge_rate(self, coulumb_val, verbosity=False):
+        if(self.abnormal_value(coulumb_val)):
+            self.print_msg("Charge rate is out of range!", verbosity)
+            return False
+        elif(self.check_in_range(coulumb_val)):
```

Comment:
An `else` will suffice here, since this is exactly the opposite of the first check?

Reply:
Yes Sir, indeed a good catch I need not check every opposite case.  It could be generalised.